lib/mongo/client.rb

```diff
@@ -1311,11 +1319,20 @@ def valid_compressors(compressors)
                        "This compressor will not be used.")
           false
         else
+
           true
         end
       end
     end
 
+    def validate_snappy_compression!
+      require 'snappy'
```

Comment:
I read about `require` calls in Ruby being somewhat slow, as an optimization this could check if the `Snappy` constant is defined prior to calling `require`.